src/main/daml/Daml/Finance/Interface/Claims/Util/Lifecycle.daml

```diff
@@ -74,6 +75,19 @@ lifecycleClaims observableCids acquisitionTime claims events = do
   observe <- collectObservables observableCids
   fmap mconcat . unzip <$> mapA (applyEvents observe acquisitionTime events) claims
 
+-- | Net claims that have the same instrument (regardless of tags)
+net : [Pending] -> [Pending]
+net pending =
+  let
+    groupedPending = groupOn (.instrument) pending
+  in
+    map (\p -> Pending with
+        tag = "netted " <> foldl (\acc e -> "|" <> e.tag <> acc) "" p
```

Comment:
I think a single fold accumulating both the `amount` and the `tag` should do it. This way we don't have to traverse the list twice

Reply:
You could also consider using [foldl1](https://docs.daml.com/daml/stdlib/DA-Foldable.html#function-da-foldable-foldl1-10610) instead of `foldl` given that you are on a non-empty structure

Reply:
> I think a single fold accumulating both the `amount` and the `tag` should do it. This way we don't have to traverse the list twice

You mean one `fold` before the `map` into a joint structure and then untangling it in the `map`, or one `fold` replacing both `folds` and the `map`?

Reply:
Inside the `map` you call `foldl` twice on the same list. The same process can be done using only one `foldl`.

Reply:
I see. Not sure if that is possible from the lambda function, but I anyway split it out to a separate named function. I think it makes it a bit more readable as well.